### DISCUSSION TITLE
Change definition of ClassProp

### DISCUSSION THREAD

**Discussion by Austaras** [swc-project/swc#3025] [General]
Change `key` field to `PropName` and remove `computed`. There's two reasons:
1. ClassProp key and ClassMethod key are both `ClassElementName` in spec, it's better to unify them
![图片](https://user-images.githubusercontent.com/15013925/145755879-8732dcce-d492-43e7-893c-e6fe1c96c926.png)
![图片](https://user-images.githubusercontent.com/15013925/145755905-c663151d-37ca-419d-b031-f0cdf89c4840.png)
2. Current definition may generate invalid state. When `computed` is false, `key` could only be `Number` `String` `BigInt`

Do we need to wait for a new major release to fix this?

**Comment by kdy1** [comment]
No, I think it's fine.
AST cannot be totally stabilized anyway. I'll mark the change as a breaking change for rust side.
